File: 0410-split-array-largest-sum/0410-split-array-largest-sum.cpp
class Solution {
public:
    bool feasible(long long sum,const vector<int>& v,int k){
        long long total=0;
        int count=1;
        for(int i:v){
            total+=i;
            if(total>sum){
                total=i;
                count++;
                if(count>k)return false;
            }
        }
        return true;
    }
    int splitArray(vector<int>& nums, int k) {
        long long right=accumulate(nums.begin(),nums.end(),0);
        long long left=*max_element(nums.begin(),nums.end());
        while(left<right){
            long long mid=left+(right-left)/2;
            if(feasible(mid,nums,k)){
                right=mid;
            }
            else left=mid+1;
        }
        return left;
    }
};

File: 0410-split-array-largest-sum/0410-split-array-largest-sum.java
class Solution {
    public  boolean feasible(int cap,int[] nums,int k){
        int tmp_cap=0,cnt=1;
        for(int i:nums){
            tmp_cap+=i;
            if(tmp_cap>cap){
                tmp_cap=i;
                cnt++;
            }
        }
        if(cnt>k)return false;
        return true;
    }
    public int splitArray(int[] nums, int k) {
        int left=nums[0],right=0;
        for(int i:nums){right+=i;
                        left=Math.max(left,i);}
        while(left<right){
            int mid=left+(right-left)/2;
            if(feasible(mid,nums,k)==false)left=mid+1;
            else right=mid;
        }
        return left;
    }
}